### DISCUSSION TITLE
Make chrome helper function works with other chromium browsers

### DISCUSSION THREAD

**Discussion by nadeemkhedr** [johnlindquist/kit#418] [Ideas]
I am currently using Brave, and for  most part, both chrome & brave **AppleScripts** are exactly the same
It would be nice to have helper functions like `focusTab`, `getActiveTab` to work with `Brave` instead of chrome

A solution might be, add an env var to set the chromium browser, by default it is set to `chrome`

**Comment by johnlindquist** [comment]
The browser can passed to the methods. The param just defaults to "Google Chrome":

```js
await getActiveTab("Brave")
await getTabs("Brave")
await focusTab("scriptkit.com", "Brave")
```

**Reply by nadeemkhedr** [reply]
Nice! didn't know that

**Reply by juanvera** [reply]
How can getActiveTab work with Arc Browser, it is a chromium one but I'm not able to locate how to name it.

**Reply by sjotterman** [reply]
Piggybacking off of this, is there something special I need to do to get these commands to work with Firefox? They work fine if I pass in 'Google Chrome', or leave the argument undefined, but they don't seem to do anything if I pass in 'Firefox'. Both browsers are installed and currently open.

**Reply by johnlindquist** [reply]
@sjotterman I found a snippet on this article:

https://matthewbilyeu.com/blog/2018-08-24/getting-url-and-tab-title-from-firefox-with-applescript

Around brought it into a script:

```js
// Name: Get Firefox Tabs

import "@johnlindquist/kit"

let result = await applescript(`
use scripting additions
use framework "Foundation"

tell application "Firefox" to activate

-- get the tab title from FF
tell application "System Events" to tell process "firefox"
	set frontmost to true
	set the_title to name of windows's item 1
	set the_title to (do shell script "echo " & quoted form of the_title & " | tr '[' ' '")
	set the_title to (do shell script "echo " & quoted form of the_title & " | tr ']' ' '")
end tell

set thePasteboard to current application's NSPasteboard's generalPasteboard()
set theCount to thePasteboard's changeCount()

-- send cmd+l and cmd+c keystrokes to FF to highlight and copy the URL
tell application "System Events"
	keystroke "l" using {command down}
	delay 0.2
	keystroke "c" using {command down}
end tell

-- wait for the clipboard content change to have been detected
repeat 20 times
	if thePasteboard's changeCount() is not theCount then exit repeat
	delay 0.1
end repeat

-- get the clipboard contents
set the_url to the clipboard

return the_url as text

`)

inspect(result)
```

@juanvera I don't have Arc, but maybe you can tweak the script above?

**Reply by sjotterman** [reply]
Thanks, @johnlindquist! This is a great jumping off point.
